pytorch_lightning/utilities/cloud_io.py

```diff
@@ -12,15 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import io
 from distutils.version import LooseVersion
 from pathlib import Path
 from typing import IO, Union
 
 import fsspec
+from fsspec.implementations.local import LocalFileSystem
+
 import torch
 
 
+class LightningLocalFileSystem(LocalFileSystem):
+    def isdir(self, path):
+        try:
+            if self.info(path)["type"] == "directory":
+                return True
+            elif self.info(path)["type"] == "link":
+                return os.path.isdir(path)
+            else:
+                return False
+        except IOError:
+            return False
+
```

Comment:
Suggest to only test for "link" and delegate all other cases to super().isdir()

Reply:
@jowagner That's a great suggestion, no need for the disclaimer! All done :smiley:

Reply:
On second thought, can actually just use `os.path.isdir` (which is what `.info` uses anyway, only it checks that first then defers to isdir otherwise).